### PR TITLE
feat: add notification background service

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

**Review by emalfroy — APPROVED** [review]

File: src/TicketingService.Monitoring/NotificationBackgroundService.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                .ToList();
+
+            var numberOfTickets = tickets
+                .Count(t =>
+                    t.Metadata.TryGetValue(MetaDataConstants.Registry, out var registry)
+                    && _options.WhitelistedRegistries.Contains(registry)
+                    && (!t.Metadata.TryGetValue(MetaDataConstants.Action, out var action)
+                        || !_options.BlacklistedActions.Contains(action)));
+
+            _notificationService
```
Comment (emalfroy) [inline src/TicketingService.Monitoring/NotificationBackgroundService.cs:69]: if(numberoftickets > 0)

File: src/TicketingService.Monitoring/NotificationBackgroundService.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                    "OpenTickets",
+                    $"{numberOfTickets} tickets not completed after {PollingInterval:g}.",
+                    "Ticketing Monitor",
+                    NotificationSeverity.Danger))
+                .Wait();
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+           _logger.LogInformation("Stopping NotificationBackgroundService");
```
Comment (ArneD) [inline src/TicketingService.Monitoring/NotificationBackgroundService.cs:80]: `_timer?.Change(Timeout.Infinite, 0);`

File: src/TicketingService.Monitoring/NotificationBackgroundService.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+           return Task.CompletedTask;
+        }
+
+        public void Dispose() { }
```
Comment (ArneD) [inline src/TicketingService.Monitoring/NotificationBackgroundService.cs:84]: `_timer?.Dispose();`

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/TicketingService.Monitoring/NotificationBackgroundService.cs (hunk 1 above)
Comment (jvandaal) [reply src/TicketingService.Monitoring/NotificationBackgroundService.cs:69]: I was about to right that, but got distracted because of the discussion...

**Pull request merged**